iothub/device/src/ClientFactory.cs

```diff
@@ -79,6 +79,14 @@ public static InternalClient Create(string hostname, string gatewayHostname, IAu
                 throw new ArgumentNullException(nameof(authenticationMethod));
             }
 
+            if (transportType != TransportType.Amqp_Tcp_Only
+                && transportType != TransportType.Mqtt_Tcp_Only
+                && authenticationMethod is DeviceAuthenticationWithX509Certificate
+                && ((DeviceAuthenticationWithX509Certificate)authenticationMethod).ChainCertificates != null)
+            {
+                throw new ArgumentException("Certificate chains are only supported on Amqp_Tcp_Only and Mqtt_Tcp_Only");
```

Comment:
so certificate chains are not supported in HTTPS and websockets? is that a hub restriction ?

Reply:
It was surely not working in websockets. I did not see HTTPS used for our device client. It seemed like it was only for file uploads so I did not test that a lot. If we use HTTPS for device client, let me know and I can do more testing and see if I can include in the next PR.

Reply:
@abhipsaMisra  to comment.

Reply:
Yes, we do use https for a small subset of device client operations - send d2c message, receive c2d message and file uploads.

Reply:
Methods and twin operations are not supported over https since these operations require a persistent connection to receive service initiated requests (for methods, twin updates etc), and https does not maintain persistent connections.
Technically we could have updated reported properties and fetched the twin propeties over https, but I guess the service team felt that the effort was not worth it.

Reply:
I will see what can be done with HTTPS after I complete this and a sample. If it is possible, I'll add support for it.